### ISSUE TITLE
Doc does not works correctly

### ISSUE THREAD

**Issue by MarcAntoine-Arnaud** [cenotelie/cratery#24]
I have a crate (`CRATE_2`) that depends of an another crate (`CRATE_1`) on the Cratery registry.

```
-- stderr
    Updating crates.io index
    Updating `sparse+https://crates.mydomain.com/` index
error: failed to get `CRATE_1` as a dependency of package `CRATE_2 v0.5.0 (/tmp/CRATE_2_0.5.0/CRATE_2-0.5.0)`

Caused by:
  no token found for `sparse+https://crates.mydomain.com/`
  consider setting up an alternate registry in Cargo's configuration
  as described by https://doc.rust-lang.org/cargo/reference/registries.html

  [registries]
  my-registry = { index = "sparse+https://crates.mydomain.com/" }
```

it seams the registry is not well defined.

**Comment by woutersl** [comment]
This is supposed to work and indeed used to work. I suppose this is a regression on `main`, maybe caused by the writing of the local configuration for git and cargo on startup. I'll check that.

**Comment by woutersl** [comment]
This fails because the credentials configuration written to disk for cargo does not include 'sparse'. We should also write the same configuration for sparse if enabled in the configuration.

**Comment by woutersl** [comment]
this should be fixed in main and version 1.10.1
